packages/expo-system-ui/android/src/main/java/expo/modules/systemui/SystemUIModule.kt

```diff
@@ -0,0 +1,262 @@
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        when (visibility) {
+          "visible" -> {
+            controller.show(WindowInsetsCompat.Type.statusBars())
+          }
+          "hidden" -> {
+            controller.hide(WindowInsetsCompat.Type.statusBars())
+            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+          }
+        }
+      }
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun getAppearance(promise: Promise) {
+    activity.runOnUiThread {
+      val appearance = when (AppCompatDelegate.getDefaultNightMode()) {
+        AppCompatDelegate.MODE_NIGHT_NO -> "light"
+        AppCompatDelegate.MODE_NIGHT_YES -> "dark"
+        AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM -> "auto"
+        else -> "default"
+      }
+      promise.resolve(appearance)
+    }
+  }
+
+  @ExpoMethod
+  fun setAppearance(appearance: String, promise: Promise) {
+    // FIXME(Marc): This crashes the App with an error from RNScreens:
+    //    java.lang.IllegalStateException: Screen fragments should never be restored.
+    //    Follow instructions from https://github.com/software-mansion/react-native-screens/issues/17#issuecomment-424704067 to properly configure your main activity.
+    //  even though the Main Activity is configured correctly (`super.onCreate(null)`)
+    //  as soon as this is fixed, we can expose it to JS again.
+    activity.runOnUiThread {
+      when (appearance) {
+        "light" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+        "dark" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+        "auto" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+        "default" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_UNSPECIFIED)
+      }
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun setDrawsBehindSystemUI(drawsBehindSystemUI: Boolean, promise: Promise) {
+    activity.runOnUiThread {
+      WindowCompat.setDecorFitsSystemWindows(activity.window, !drawsBehindSystemUI)
+      promise.resolve(null)
+    }
+  }
+
+  @ExpoMethod
+  fun getDrawsBehindSystemUI(promise: Promise) {
+    activity.runOnUiThread {
+      promise.resolve(!activity.window.decorView.fitsSystemWindows)
+    }
+  }
+
+  @ExpoMethod
+  fun setSystemUIBehavior(behavior: String, promise: Promise) {
+    activity.runOnUiThread {
+      WindowInsetsControllerCompat(activity.window, activity.window.decorView).let { controller ->
+        controller.systemBarsBehavior = when (behavior) {
+          "overlay-swipe" -> WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
+          "inset-swipe" -> WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_SWIPE
+          "inset-touch" -> WindowInsetsControllerCompat.BEHAVIOR_SHOW_BARS_BY_TOUCH
```

Comment:
Maybe we shouldn't add this since it's [going to be deprecated](https://developer.android.com/reference/android/view/WindowInsetsController#BEHAVIOR_SHOW_BARS_BY_TOUCH).

Reply:
It's being deprecated in API 31, min API is still 21 (or something like that)

Reply:
well it seems like it's being deprecated because there's a replacement value, so if that replacement is available on 21, then we should use that and not introduce the deprecated value imo.